spec/unit/auto_registrar_spec.rb

```diff
@@ -22,6 +23,17 @@
     expect(container["foo"]).to be_an_instance_of(Test::Foo)
     expect(container["bar"]).to be_an_instance_of(Test::Bar)
     expect(container["bar.baz"]).to be_an_instance_of(Test::Bar::Baz)
+
+    expect { container["bar.abc"] }.to raise_error(
+      Dry::System::AutoRegisterableComponentNotLoadable
+    ) do |error|
+      expect(error.message).to start_with(
+        "Component 'bar.abc' with auto_register: true is not loadable. "\
+        "Looking for Test::Bar::Abc. You may need customize your inflector.\n"\
+      ).and end_with(
+        "Did you mean?  Test::Bar::ABC"
+      )
```

Comment:
Yes, thanks for this!

Defining our own Checker doesn't really do much, since the message formatting is elsewhere. Instead, we can just use the existing checker's `corrections` array to get what we're looking for, then constructing our own string based on that.